src/Csrf/CsrfViewInjection.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Yiisoft\Yii\View\Csrf;
+
+use Yiisoft\Router\UrlMatcherInterface;
+use Yiisoft\Yii\View\ContentParamsInjectionInterface;
+use Yiisoft\Yii\View\LayoutParamsInjectionInterface;
+use Yiisoft\Yii\View\MetaTagsInjectionInterface;
+
+class CsrfViewInjection implements
+    ContentParamsInjectionInterface,
+    LayoutParamsInjectionInterface,
+    MetaTagsInjectionInterface,
+    CsrfViewInjectionInterface
+{
+    public const DEFAULT_REQUEST_ATTRIBUTE = 'csrf_token';
+    public const DEFAULT_META_ATTRIBUTE = 'csrf';
+    public const DEFAULT_PARAMETER = 'csrf';
+
+    private UrlMatcherInterface $urlMatcher;
+
+    private string $requestAttribute = self::DEFAULT_REQUEST_ATTRIBUTE;
+    private string $metaAttribute = self::DEFAULT_META_ATTRIBUTE;
+    private string $parameter = self::DEFAULT_PARAMETER;
+
+    private ?string $csrfToken = null;
+
+    public function __construct(UrlMatcherInterface $urlMatcher)
+    {
+        $this->urlMatcher = $urlMatcher;
+    }
+
+    public function withRequestAttribute(?string $requestAttribute = null): CsrfViewInjectionInterface
+    {
+        $clone = clone $this;
+        $clone->requestAttribute = $requestAttribute ?? self::DEFAULT_REQUEST_ATTRIBUTE;
+        return $clone;
+    }
+
+    public function withParameter(string $parameter): self
```

Comment:
It's not clear from the name that this is to set parameter name and not the value.

Reply:
Instead `parameter` use `parameterName` in function and variable names?